Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -162,3 +162,109 @@ instance : HasForget₂ ProfiniteGrp Grp where
+  mul_mem' hx hy _ _ π := by simp only [Pi.mul_apply, map_mul, hx π, hy π]
+  one_mem' := by simp only [Set.mem_setOf_eq, Pi.one_apply, map_one, implies_true]
+  inv_mem' h _ _ π := by simp only [Pi.inv_apply, map_inv, h π]
+
+/--The functor mapping a profinite group to its underlying profinite space-/
+def profiniteGrpToProfinite : ProfiniteGrp ⥤ Profinite where
+  obj G := G.toProfinite
+  map f := ⟨f, by continuity⟩
+
+instance instProfiniteGrpToProfiniteFaithful : profiniteGrpToProfinite.Faithful := {
+  map_injective := fun {_ _} _ _ h =>
+    ConcreteCategory.hom_ext_iff.mpr (congrFun (congrArg ContinuousMap.toFun h)) }
+
+instance : CompactSpace (limit F) := inferInstanceAs
+  (CompactSpace (Profinite.limitCone (F ⋙ profiniteGrpToProfinite.{max v w'})).pt)
+
+/-- Making the direct limit of `FiniteGrp` into a `ProfiniteGrp`. -/
+def ofLimit : ProfiniteGrp := .of (ProfiniteGrp.limit F)
+
+/-- Verify that the limit constructed above exist projections to the `FiniteGrps`
+that are compatible with the morphisms between them. -/
+def limitCone : Limits.Cone F where
+  pt := ofLimit F
+  π :=
+  { app := fun j => {
+      toFun := fun x => x.1 j
+      map_one' := rfl
+      map_mul' := fun x y => rfl
+      continuous_toFun := by
+        exact (continuous_apply j).comp (continuous_iff_le_induced.mpr fun U a => a)
+    }
+    naturality := by
+      intro i j f
+      simp only [Functor.const_obj_obj, Functor.comp_obj,
+        Functor.const_obj_map, Category.id_comp, Functor.comp_map]
+      congr
+      exact funext fun x => (x.2 f).symm
+  }
+
+@[simp]
+lemma limitCone_pt : (ProfiniteGrp.limitCone F).pt =
+    ProfiniteGrp.ofLimit F := rfl
+
+@[simp, nolint simpNF]
+lemma limitCone_π_app_apply  (j : J) (x : ofLimit F) :
+    ((ProfiniteGrp.limitCone F).π.app j) x = x.1 j := rfl
+
+/-- Verify that the limit constructed above satisfies the universal property. -/
+def limitConeIsLimit : Limits.IsLimit (limitCone F) where
+  lift cone := {
+    toFun := ((Profinite.limitConeIsLimit (F ⋙ profiniteGrpToProfinite)).lift
+      (profiniteGrpToProfinite.mapCone cone)).toFun
+    map_one' := by
+      apply SetCoe.ext
+      ext j
+      exact map_one (cone.π.app j)
+    map_mul' := fun _ _ ↦ by
+      apply SetCoe.ext
+      ext j
+      exact map_mul (cone.π.app j) _ _
+    continuous_toFun := ((Profinite.limitConeIsLimit (F ⋙ profiniteGrpToProfinite)).lift
+      (profiniteGrpToProfinite.mapCone cone)).continuous }
+  uniq cone m h := by
+    apply instProfiniteGrpToProfiniteFaithful.map_injective
+    simpa using (Profinite.limitConeIsLimit (F ⋙ profiniteGrpToProfinite)).uniq
+      (profiniteGrpToProfinite.mapCone cone) (profiniteGrpToProfinite.map m)
+      (fun j ↦ congrArg profiniteGrpToProfinite.map (h j))
+
+@[simp, nolint simpNF]
+lemma limitConeIsLimit_lift_toFun_coe (j : J) (cone : Limits.Cone F)
+    (pt : ↑cone.pt.toProfinite.toTop) :
+    (((ProfiniteGrp.limitConeIsLimit F).lift cone) pt).val j = (cone.π.app j) pt := rfl
+
```

Comment:
I'm not sure this is a good `simp` lemma. Generally, you'd want to use the API for `IsLimit` instead of definitional properties of a limit cone.

Reply:
This is generated by simps, because of the normal form problem it is moved out side. maybe the simps isn't good at all

Reply:
I would just remove this lemma if you don't have a specific use for it in mind